### ISSUE TITLE
Add Swatch Data to Aggregation Options

### ISSUE THREAD

**Issue by t-heuser** [magento/magento2#33269]
*As a developer I want to be able to receive the swatch data for an AggregationOption so that I can show the customer images/colors for filters.*

The type "AggregationOption" currently only returns the value and a label. I want it to also to return" swatch_data" with the type "SwatchData".  
Previously it was possible to reveive this data when searching for products as the type "Products" has the field "filters" which leads to the "swatch_data" in the type "SwatchLayerFilterItem". But the field "filters" is deprecated now and "aggregations" is the type to be used. But the "swatch_data" information cannot be obtained now because of the deprecations.

**Suggested solution**

The solution would be to implement a new type "SwatchAggregationOption" which implements the "AggregationOptionInterface" and has the extra field "swatch_data" which returns the type "SwatchDataInterface".

**Comment by karyna-tsymbal-atwix** [comment]
@magento I am working on this

**Comment by engcom-Hotel** [comment]
Hi @oneserv-heuser,

Thank you for reporting this.

Based on Docs [here](https://devdocs.magento.com/guides/v2.4/graphql/queries/products.html). I can notice that ‘Swatch Image’ is ‘Deprecated input attributes’ and it can’t be used anymore.

We can’t consider this issue as a bug since that functionality attribute is deprecated. But users may require to use that option so we can consider it as ‘Feature Request’ so moving to the same Board where it can be further analyzed and necessary actions can be taken.

Thanks

**Comment by oneserv-heuser** [comment]
@engcom-Hotel You're totally right, this was never intended to be a bug request. It's a missing feature which existed before but is now gone because of the deprecation of the field "filters".
